testsuite/tests/typing-modules/functors.ml

```diff
@@ -0,0 +1,1189 @@
+               (Z : functor (A : sig type za end) (B : sig type zbb end) ->
+                      sig end)
+               -> sig end
+         end
+       is not included in
+         sig
+           module F :
+             functor
+               (X : functor (A : sig type xa end) (B : sig type xz end) ->
+                      sig end)
+               (Y : functor (A : sig type ya end) (B : sig type yb end) ->
+                      sig end)
+               (Z : functor (A : sig type za end) (B : sig type zb end) ->
+                      sig end)
+               -> sig end
+         end
+       In module F:
+       Modules do not match:
+         functor (X : ...(X)) (Y : ...(Y)) (Z : ...(Z)) -> ...
+       is not included in
+         functor (X : ...(X)) (Y : ...(Y)) (Z : ...(Z)) -> ...
+  1. Module types do not match:
+       ...(X) =
+       functor (A : sig type xaa end) (B : sig type xz end) -> sig end
+     does not include
+       ...(X) =
+       functor (A : sig type xa end) (B : sig type xz end) -> sig end
+     Modules do not match:
+       functor (A : ...(A)) (B : ...(B)) -> ...
+     is not included in
+       functor (A : ...(A)) (B : ...(B)) -> ...
+     1. Module types do not match:
+          ...(A) = sig type xa end
+        does not include
+          ...(A) = sig type xaa end
+        The type `xa' is required but not provided
+     2. Module types ...(B) and ...(B) match
+  2. Module types do not match:
+       ...(Y) =
+       functor (A : sig type ya end) (B : sig type ybb end) -> sig end
+     does not include
+       ...(Y) =
+       functor (A : sig type ya end) (B : sig type yb end) -> sig end
+     Modules do not match:
+       functor (A : ...(A)) (B : ...(B)) -> ...
+     is not included in
+       functor (A : ...(A)) (B : ...(B)) -> ...
+  3. Module types do not match:
+       ...(Z) =
+       functor (A : sig type za end) (B : sig type zbb end) -> sig end
+     does not include
+       ...(Z) =
+       functor (A : sig type za end) (B : sig type zb end) -> sig end
+     Modules do not match:
+       functor (A : ...(A)) (B : ...(B)) -> ...
+     is not included in
+       functor (A : ...(A)) (B : ...(B)) -> ...
+|}]
+
+module A: sig
+  module B: sig
+    module C: sig
+      module D: sig
+        module E: sig
+          module F: sig type x end -> sig type y end
+          -> sig type z end -> sig type w end -> sig end
+        end
+      end
+    end
+  end
+end = struct
+  module B = struct
+    module C = struct
+      module D = struct
+        module E = struct
+          module F(X:sig type x end)(Y:sig type y' end)
+            (W:sig type w end) = struct end
+        end
+      end
+    end
+  end
+end
+[%%expect {|
+Lines 12-23, characters 6-3:
+12 | ......struct
+13 |   module B = struct
+14 |     module C = struct
+15 |       module D = struct
+16 |         module E = struct
+...
+20 |       end
+21 |     end
+22 |   end
+23 | end
+Error: Signature mismatch:
+       Modules do not match:
```

Comment:
Do you really need to repeat all the steps here?

Reply:
I intend to remove this behavior, but this is the current behavior; and I wanted to avoid changing too many parameters at the same time.